src/sentry/static/sentry/app/components/events/relatedEvents/index.tsx

```diff
@@ -0,0 +1,278 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+import uniqBy from 'lodash/uniqBy';
+import {capitalize} from 'lodash';
+
+import space from 'app/styles/space';
+import {t, tct} from 'app/locale';
+import DateTime from 'app/components/dateTime';
+import {Organization, Event, GlobalSelection} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {transactionSummaryRouteWithQuery} from 'app/views/performance/transactionSummary/utils';
+import EventDataSection from 'app/components/events/eventDataSection';
+import {PanelTable, PanelItem} from 'app/components/panels';
+import PlatformIcon from 'app/components/platformIcon';
+import Link from 'app/components/links/link';
+import Button from 'app/components/button';
+import DiscoverQuery, {TableDataRow} from 'app/utils/discover/discoverQuery';
+import Tooltip from 'app/components/tooltip';
+import {generateEventSlug, eventDetailsRouteWithEventView} from 'app/utils/discover/urls';
+import {DEFAULT_RELATIVE_PERIODS} from 'app/constants';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import TimeSince from 'app/components/timeSince';
+import {IconClock, IconFire, IconSpan} from 'app/icons';
+
+enum CURRENT_LOCATION {
+  DISCOVER = 'discover',
+  ISSUES = 'issues',
+  PERFORMANCE = 'performance',
+}
+
+enum EVENT_TYPE {
+  ERROR = 'error',
+  TRANSACTION = 'transaction',
+}
+
+type Props = {
+  organization: Organization;
+  event: Event;
+  location: Location;
+  selection: GlobalSelection;
+};
+
+// List events that have the same tracing ID as the current Event
+const RelatedEvents = ({organization, location, event, selection}: Props) => {
+  const orgSlug = organization.slug;
+  const orgFeatures = new Set(organization.features);
+
+  const getCurrentLocation = () => {
+    const pathname = location.pathname.split('/');
+    switch (pathname[3]) {
+      case 'discover':
+        return CURRENT_LOCATION.DISCOVER;
+      case 'issues':
+        return CURRENT_LOCATION.ISSUES;
+      default:
+        return CURRENT_LOCATION.PERFORMANCE;
+    }
+  };
+
+  const getEventView = () => {
+    // traceId should always be defined
+    const traceId = event.contexts?.trace?.trace_id;
```

Comment:
While the tracing context should be there, it won't always be there if customers are using old SDK's, or SDKs on native where there isn't any tracing support.